tests/unit/test_errorhandler.py

```diff
@@ -10,101 +10,74 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
-from awscli.testutils import unittest
+import io
+from collections import namedtuple
+
+from botocore.exceptions import (
+    NoRegionError, NoCredentialsError, ClientError,
+    ParamValidationError as BotocoreParamValidationError,
+)
+from awscli.arguments import UnknownArgumentError
+from awscli.argparser import ArgParseException
+from awscli.argprocess import ParamError, ParamSyntaxError
+from awscli.customizations.exceptions import (
+    ParamValidationError, ConfigurationError
+)
 
-import mock
+from awscli.testutils import unittest
 from awscli import errorhandler
 
 
-class TestErrorHandler(unittest.TestCase):
+class TestChainedExceptionHandler(unittest.TestCase):
```

Comment:
Then if we move to using `yield` we can just get rid of this class and replace the `setUpClass` method with calling `construct_default_cli_error_handlers_chain` in the `_assert_rc_and_error_message` function.